backend/inference.py

```diff
@@ -33,12 +33,11 @@ def process_inference(
             task_id,
             client_id):
 
-        # Parse the JSON string
-        data = json.loads(inference.to_string())
+        # Convert the string to a list of integers
+        inference_list = [int(x) for x in inference.to_string().strip('[]').split(',')]
 
```

Comment:
Plase try something similar to this:

```suggestion
        inference_str = inference.to_string().strip('[]')

        inference_json = json.loads(inference_str)

        # Extract the 'data' field which contains the list of numbers
        inference_list = inference_json.get('data', [])

        # Convert each value in the list to a float or integer
        inference_list = [int(x) for x in inference_list]
```